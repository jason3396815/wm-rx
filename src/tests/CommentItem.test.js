import { h, Component } from 'preact';
import CommentItem from '../components/CommentItem';

describe('<CommentItem />', () => {
	it('should render a comment item', () => {
		expect(
			<CommentItem comment={{
				level: 0,
				time_ago: '3 days ago',
				user: 'vutran',
				comments: [],
				content: 'Test body',
			}} />
		).toMatchSnapshot();
	});

	it('should render a comment replies', () => {
		expect(
			<CommentItem comment={{
				level: 0,
				time_ago: '3 days ago',
				user: 'vutran',
				comments: [
					{
						level: 1,
						time_ago: '2 days ago',
						user: 'tranvu',
						comments: [],
						content: 'Test reply body',
					},
				],
				content: 'Test body',
			}} />
		).toMatchSnapshot();
	});
});


