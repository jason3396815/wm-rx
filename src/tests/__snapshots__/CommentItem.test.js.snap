// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentItem /> should render a comment item 1`] = `
e {
  "attributes": Object {
    "comment": Object {
      "comments": Array [],
      "content": "Test body",
      "level": 0,
      "time_ago": "3 days ago",
      "user": "vutran",
    },
  },
  "children": Array [],
  "key": undefined,
  "nodeName": [Function],
}
`;

exports[`<CommentItem /> should render a comment replies 1`] = `
e {
  "attributes": Object {
    "comment": Object {
      "comments": Array [
        Object {
          "comments": Array [],
          "content": "Test reply body",
          "level": 1,
          "time_ago": "2 days ago",
          "user": "tranvu",
        },
      ],
      "content": "Test body",
      "level": 0,
      "time_ago": "3 days ago",
      "user": "vutran",
    },
  },
  "children": Array [],
  "key": undefined,
  "nodeName": [Function],
}
`;
